packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -11,10 +11,707 @@ import 'picker.dart';
+          longestText = localizations.datePickerHour(i);
+    }
+    else if (columnType == 'm') {
+      for (int i = 0 ; i < 60; i++)
+        if (longestText.length < localizations.datePickerMinute(i).length)
+          longestText = localizations.datePickerMinute(i);
+    }
+    else
+      longestText =
+        localizations.anteMeridiemAbbreviation.length > localizations.postMeridiemAbbreviation.length
+        ? localizations.anteMeridiemAbbreviation
+        : localizations.postMeridiemAbbreviation;
+
+    final TextPainter painter = new TextPainter(
+      text: TextSpan(
+        style: DefaultTextStyle.of(context).style,
+        text: longestText,
+      ),
+      textDirection: Directionality.of(context),
+    );
+    painter.layout();
+
+    return painter.maxIntrinsicWidth;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Widths of the columns in this picker, ordered from left to right.
+    final List<double> columnWidth = <double>[
+      _getPickerWidth('h'),
+      _getPickerWidth('m'),
+    ];
+    final List<Function> pickerBuilder = <Function>[
+      _buildHourPicker,
+      _buildMinutePicker,
+    ];
+
+    // Adds am/pm column if the picker is not using 24h format.
+    if (!widget.use24hFormat) {
+      if (localizations.datePickerDateTimeOrder == DatePickerDateTimeOrder.date_time_dayPeriod
+        || localizations.datePickerDateTimeOrder == DatePickerDateTimeOrder.time_dayPeriod_date) {
+        pickerBuilder.add(_buildAmPmPicker);
+        columnWidth.add(_getPickerWidth('a'));
+      }
+      else {
+        pickerBuilder.insert(0, _buildAmPmPicker);
+        columnWidth.insert(0, _getPickerWidth('a'));
+      }
+    }
+
+    // Adds medium date column if the picker's mode is date and time.
+    if (widget.mode == CupertinoDatePickerMode.dateAndTime) {
+      if (localizations.datePickerDateTimeOrder == DatePickerDateTimeOrder.time_dayPeriod_date
+          || localizations.datePickerDateTimeOrder == DatePickerDateTimeOrder.dayPeriod_time_date) {
+        pickerBuilder.add(_buildMediumDatePicker);
+        columnWidth.add(_getPickerWidth('d'));
+      }
+      else {
+        pickerBuilder.insert(0, _buildMediumDatePicker);
+        columnWidth.insert(0, _getPickerWidth('d'));
+      }
+    }
+
+    final List<Widget> children = <Widget>[];
+
+    for (int i = 0; i < columnWidth.length; i++) {
+      double offAxisFraction = 0.0;
+      if (i == 0)
+        offAxisFraction = -0.5 * textDirectionFactor;
+      else if (i >= 2 || columnWidth.length == 2)
+        offAxisFraction = 0.5 * textDirectionFactor;
+
+      EdgeInsets padding = const EdgeInsets.only(right: _kDatePickerPadSize);
+      if (i == columnWidth.length - 1)
+        padding = padding.flipped;
+      if (textDirectionFactor == -1)
+        padding = padding.flipped;
+
+      children.add(LayoutId(
+        id: i,
+        child: pickerBuilder[i](
+          offAxisFraction,
+          (Widget child) {
```

Comment:
Can't you make this an instance method so you don't have to keep creating new closure instances?